src/components/StringFormatter/__tests__/StringFormatter.spec.js

```diff
@@ -3,34 +3,70 @@
  * @copyright IBM Security 2019
  */
 
-import { shallow } from 'enzyme';
+import { render } from '@testing-library/react';
 import React from 'react';
 
 import { StringFormatter } from '../../..';
-import value from '../_mocks_';
+import { namespace } from '../StringFormatter';
 
 describe('StringFormatter', () => {
-  let stringFormatter;
-
-  beforeEach(() => {
-    stringFormatter = shallow(<StringFormatter value={value} />);
+  test('should have no Axe or DAP violations', async () => {
+    const main = document.createElement('main');
+    render(
+      <StringFormatter
+        value="This is a long test string that would normally be truncated."
+        truncate
+        lines={2}
+        width="50px"
+      />,
+      {
+        // DAP requires a landmark '<main>' in the DOM:
+        container: document.body.appendChild(main),
+      }
+    );
+    await expect(document.body).toHaveNoAxeViolations();
+    await expect(document.body).toHaveNoDAPViolations('StringFormatter');
   });
 
-  it('renders', () => {
-    expect(stringFormatter).toMatchSnapshot();
+  test('should add a custom class', () => {
+    const { getByText } = render(
+      <StringFormatter value="test content" className="custom-class" />
+    );
+    expect(getByText(/test content/i)).toHaveClass('custom-class');
   });
 
-  it('renders the truncated variation', () => {
-    stringFormatter.setProps({ truncate: true });
+  test('should truncate text via the `truncate` prop', () => {
+    const { getAllByText } = render(
+      <StringFormatter value="test content" truncate />
+    );
+    // Expect two elements with the target text: the base component
+    // and the tooltip that is added when `truncate={true}`.
+    expect(getAllByText(/test content/i).length).toBe(2);
 
-    expect(stringFormatter).toMatchSnapshot();
+    // Expect to see the truncate class on the inner non-tooltip element:
+    expect(getAllByText(/test content/i)[0]).toHaveClass(
+      `${namespace}--truncate`
+    );
   });
 
-  it('renders the correct amount of lines', () => {
-    stringFormatter.setProps({ truncate: true, lines: 4 });
-    expect(stringFormatter.find('span').prop('style')).toHaveProperty(
-      'WebkitLineClamp',
-      4
+  // Does not work:
+  // test('should apply correct style attribute when `lines` provided', () => {
+  //   const { getAllByText } = render(
+  //     <StringFormatter value="test content" lines={4} />
+  //   );
+  //   expect(getAllByText(/test content/i)[0]).toHaveAttribute(
```

Comment:
Would the [`.toHaveStyle`](https://github.com/testing-library/jest-dom#tohavestyle) matcher work?

Reply:
I tried that, but the `style` attribute doesn't even get added. 😭 If you try debugging / seeing the component render, you won't see the `style` attribute at all. I wonder if it has something to do with the `-webkit` prefix? (As far as I can tell, that is required, as `line-camp` doesn't have consistent support).

I did go ahead and update all uses of `toHaveAttribute('style', '...')` to `toHaveStyle` though -- thanks for the suggestion! I'll remember to use that one. 

I also kept the commented out test 🤔 Worthwhile to keep it there, you think?

Reply:
Yeah, I have no idea how that works. The commented test would be valuable to keep if there is an expectation that this will be resolved eventually and the test will be uncommented then. If you don't want to dig deeper into why it's not working, I would just remove it.